framework_tvb/tvb/adapters/visualizers/region_volume_mapping.py

```diff
@@ -216,12 +215,30 @@ def compute_params(self, region_mapping_volume=None, measure=None, data_slice=''
         return params
 
     def get_volume_view(self, entity_gid, **kwargs):
-        h5_class, h5_path = self.load_h5_of_gid(entity_gid)
-        with h5_class(h5_path) as h5_opened:
-            if h5_class is TimeSeriesRegionH5:
-                return self._get_volume_view_region(h5_opened, **kwargs)
+        ts_region_h5 = h5.h5_file_for_gid(entity_gid)
+        with ts_region_h5:
+            if isinstance(ts_region_h5, TimeSeriesRegionH5):
+                return self._get_volume_view_region(ts_region_h5, **kwargs)
 
-            return h5_opened.get_volume_view(**kwargs)
+            volume_view = ts_region_h5.get_volume_view(**kwargs)
+
+        return volume_view
+
+    @staticmethod
+    def prepare_space_parameters(ts_h5, x_plane, y_plane, z_plane):
+        region_mapping_volume_gid = ts_h5.region_mapping_volume.load()
+
+        if region_mapping_volume_gid is None:
+            raise Exception("Invalid method called for TS without Volume Mapping!")
+
+        volume_rm_h5 = h5.h5_file_for_gid(region_mapping_volume_gid)
+        volume_rm_shape = volume_rm_h5.array_data.shape
```

Comment:
This method is hard to follow and one could easily forget to close the `volume_rm_h5` outside it.
We could do a generic method for both visualizers that handles the opening of `volume_rm_h5` using `with` and then calls `_get_volume_view_region` which receives also `volume_rm_h5` as arg but does something different for each viewer. What do you think?

Reply:
I have done it. I needed to change more things in order to make it work, so please have a look to make sure I did not miss something.